File: jphp-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXGamePane.php
<?php
namespace php\game;

use php\gui\layout\UXAnchorPane;
use php\gui\layout\UXScrollPane;
use php\gui\paint\UXColor;
use php\gui\UXNode;

/**
 * Class UXGamePane
 * @package php\game
 */
class UXGamePane extends UXAnchorPane
{
    /**
     * @var UXGameScene
     */
    public $scene;

    /**
     * @var UXAnchorPane
     */
    public $content;

    /**
     * @var bool
     */
    public $autoSize = false;

    /**
     * @var string|UXColor
     */
    public $areaBackgroundColor = 'white';

    /**
     * @readonly
     * @var int
     */
    public $viewX = 0, $viewY = 0;

    /**
     * @var float
     */
    public $viewWidth = 0.0, $viewHeight = 0.0;

    /**
     * @param $x
     * @param $y
     */
    public function scrollTo($x, $y)
    {
    }

    /**
     * @param UXAnchorPane $area
     */
    public function loadArea(UXAnchorPane $area)
    {
    }
}

File: jphp-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXGamePane.java
package org.develnext.jphp.ext.game.classes;

import javafx.scene.Node;
import javafx.scene.layout.AnchorPane;
import javafx.scene.paint.Color;
import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.GamePane;
import org.develnext.jphp.ext.javafx.classes.layout.UXScrollPane;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Namespace(GameExtension.NS)
public class UXGamePane extends UXScrollPane<GamePane> {
    interface WrappedInterface {
        @Property("scrollY") double vvalue();
        @Property("scrollX") double hvalue();

        @Property double areaWidth();
        @Property double areaHeight();
        @Property double[] areaSize();

        @Property boolean autoSize();

        @Property Color areaBackgroundColor();

        void loadArea(AnchorPane area);
    }

    public UXGamePane(Environment env, GamePane wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGamePane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public void __construct() {
        __wrappedObject = new GamePane();
    }

    @Override
    public void __construct(@Reflection.Nullable Node content) {
        throw new IllegalStateException();
    }
}
